packages/teams-js/src/public/media.ts

```diff
@@ -408,34 +412,69 @@ export namespace media {
      * @hidden
      * Hide from docs
      * --------
+     *
+     * Function to notify the host client to programatically control the experience
+     * @param mediaEvent indicates what the event that needs to be signaled to the host client
+     * @returns A promise resolved promise
+     */
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void>;
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     *
+     * @deprecated
+     * As of 2.0.0-beta.3, please use {@link media.MediaController.notifyEventToHost media.MediaController.notifyEventToHost(mediaEvent: MediaControllerEvent): Promise\<void\>} instead.
+     *
      * Function to notify the host client to programatically control the experience
      * @param mediaEvent indicates what the event that needs to be signaled to the host client
      * Optional; @param callback is used to send app if host client has successfully handled the notification event or not
      */
-    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void {
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void;
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
       ensureInitialized(FrameContexts.content, FrameContexts.task);
-      const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
-      if (err) {
-        if (callback) {
-          callback(err);
-        }
-        return;
-      }
 
-      const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
-      sendMessageToParent('media.controller', [params], (err?: SdkError) => {
-        if (callback) {
-          callback(err);
-        }
-      });
+      return Promise.resolve()
+        .then(() => {
+          return throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion);
+        })
+        .then(() => {
+          const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+
+          sendMessageToParent('media.controller', [params], (err?: SdkError) => {
+            if (callback) {
+              callback(err);
+            } else if (err) {
+              throw err;
+            }
+          });
```

Comment:
I'm kind of confused by this, couldn't we do something like 
```
return Promise.resolve()
  .then(() => {
         const err = isMobileApiSupported(nonFullScreenVideoModeAPISupportVersion);
          if(err){
              throw err;
           }
         const params: MediaControllerParam = {...}
         ...  
    }
    .catch
```

Reply:
this way we don't have to use .then().then().catch()

Reply:
you can't assign the error to a variable because it is throwing errors now

Reply:
i can definitely put comments in there though because i was very confused by this function originally

Reply:
plus, this is handling the old use case but with promises, we need the function to exit after the first then if it throws an error, and go to the catch. This was something i was confused about in the original implementation

Reply:
Hmm so it's not possible to propagate the Error up to this call and then throw it? Basically just switching the callback in the original implementation to a throw

Reply:
it is, it's in the catch, you can't write something like if(err) because you can't assign it to a variable, it will always be void.  so i have to split that function into a couple thens because if the first then fails, it has to immediately exit the function, if it doesn't it continues, but the callback could have an error, and since we don't want to swallow that case, we need to call the callback with the error, but throw it in the case where there is no callback.  

i'd be more than happy to discuss this offline, i had to have it explained to me a couple of times :)